Emrge-2163 Adding overwrite functionality

timmockovative opened the pull request:
## What
<!--
* Describe what the change is solving. Link all GitHub issues related to this change.
-->
Adds overwrite functionality to the airbyte connector

## How
<!--
* Describe how code changes achieve the solution.
-->

## Review guide
<!--
1. `x.py`
2. `y.py`
-->

## User Impact
<!--
* What is the end result perceived by the user?
* If there are negative side effects, please list them. 
-->

## Can this PR be safely reverted and rolled back?
<!--
* If unsure, leave it blank.
-->
- [x] YES 💚
- [ ] NO ❌

timmockovative committed:
Upgrading to v0.3.9 which includes full refresh overwrite capabilities.

timmockovative committed:
Upgrading to v0.3.9 which includes full refresh overwrite capabilities.

JohnMikkelsonOvative commented on airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py:
```diff
@@ -115,17 +122,6 @@ def write_parquet(self, messages, stream_name, stream_schema):
         buf = pa.BufferOutputStream()
 
-        #build path
-        path_name = ""
```
the `path_name` variable still exists, but the 'build path' code has been deleted - how does the new write functionality allow for pathing, or does it remove it?

timmockovative replied:
Oh, moved the path building over to the stream writer file. Reduced a bit of redundant code and allowed me to use it for the file overwrite code as well. Path name now comes from the parameters

JohnMikkelsonOvative commented on airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py:
```diff
@@ -145,25 +141,14 @@ def write_csv(self, messages, stream_name, flattening):
         df.to_csv(buf)  # filling that buffer
 
-        #build path
-        path_name = ""
```
same as above on `path_name`

JohnMikkelsonOvative commented on airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py:
```diff
@@ -261,16 +246,6 @@ def reorder_list_of_dicts_by_schema(records, schema_order):
 
         schema = avro.schema.parse(schema_json)
 
-        path_name = ""
```
same comment on `path_name` functionality

timmockovative commented:
Moved pathing over from the individual write functions to the streamwriter. The path writing still follows the same structure as before. Added in overwrite functionality. If the sync mode is overwrite, then it will blast the folder location (if it has anything) on the first flush.

>
> Important: This command can only be run by one of the repository admins:
> Usage: ./tools/bin/get_repo_admins.sh <personal_access_token> <repository>
